core/trino-main/src/main/java/io/trino/sql/analyzer/Analysis.java

```diff
@@ -918,18 +919,17 @@ public void addColumnMask(Table table, String column, Expression mask)
                 .map(entry -> {
                     NodeRef<Table> table = entry.getKey();
 
-                    List<ColumnInfo> columns = referencedFields.get(table).stream()
-                            .filter(field -> field.getName().isPresent()) // For DELETE queries, the synthetic column for row id doesn't have a name
-                            .map(field -> {
-                                String fieldName = field.getName().get();
-
-                                return new ColumnInfo(
-                                        fieldName,
-                                        columnMasks.getOrDefault(table, ImmutableMap.of())
-                                                .getOrDefault(fieldName, ImmutableList.of()).stream()
-                                                .map(Expression::toString)
-                                                .collect(toImmutableList()));
-                            })
+                    QualifiedObjectName tableName = entry.getValue().getName();
+                    List<ColumnInfo> columns = tableColumnReferences.values().stream()
+                            .map(e -> e.getOrDefault(tableName, ImmutableSet.of()))
```

Comment:
nit : Instead of `getOrDefault` can we replace it with a `filter` and a `map(Map::get)`